src/bundler.js

```diff
@@ -0,0 +1,48 @@
+const fs = require('fs')
+const { basename, extname, join } = require('path')
+const { promisify } = require('util')
+
+const esbuild = require('esbuild')
+
+const pUnlink = promisify(fs.unlink)
+
+const bundleJsFile = function ({
+  additionalModulePaths,
+  destFilename,
+  destFolder,
+  externalModules = [],
+  ignoredModules = [],
+  srcFile,
+}) {
+  // De-duping external and ignored modules.
+  const external = [...new Set([...externalModules, ...ignoredModules])]
+  const jsFilename = `${basename(destFilename, extname(destFilename))}.js`
+  const bundlePath = join(destFolder, jsFilename)
+
+  // We use the synchronous entry point of the esbuild API because it's more
+  // performant, works with Node 8, and there are no obvious downsides since
+  // we don't need to be performing other work simultaneously.
```

Comment:
Isn't each function bundled separately?
In that case, use a synchronous API would bundle each function serially instead of in parallel.

Reply:
You're absolutely right, this was an oversight on my part. I was focusing on the fact that it's fine to block while the bundling is taking place, because we can't proceed with any other operation until it's complete. But I was forgetting the fact that we might actually be bundling several, independent functions at the same time, in which case spinning different processes (and utilising all the available cores) is much more performant.

Sadly, the async API doesn't work with Node 8, but rather than introducing a performance penalty for everyone, I've added a check for the version being used and switch to the sync API only for Node 8.x: https://github.com/netlify/zip-it-and-ship-it/pull/325/commits/05bedb014f2fe6ca65cdbcdeba70234b68036782.

Thanks @ehmicky!

Reply:
This looks perfect!